tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_station.py

```diff
@@ -0,0 +1,71 @@
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+from boardfarm.devices.debian_wifi import DebianWifi
+from boardfarm.devices import connection_decider
+from environment import VirtualAPHostapd
+
+
+class PrplMeshStation(DebianWifi):
+    """Client of prplMesh enabled Access Point."""
+
+    linesep = "\r"
+    model = "prplWRT_STA"
+    prompt = ['.*:~', '/.*#']
+
+    def __init__(self, *args, **kwargs):
+        """Init station and wlan iface."""
+        self.args = args
+        self.kwargs = kwargs
+
+        config = kwargs.get("config", kwargs)
+        self.connection_type = config.get("connection_type", None)
+        self.connection = connection_decider.connection(device=self,
+                                                        conn_type=self.connection_type,
+                                                        **kwargs)
+        self.connection.connect()
+        self.consoles = [self]
+        super(PrplMeshStation, self).__init__(*args, **kwargs)
+        self.iface_dut = self.iface_wifi = self.kwargs.get(
+            'iface', 'wlan0')
+        self.driver_name = config.get("driver", "nl80211,wext")
+        self.mac = self.get_mac()
+
+        self.wifi_disconnect()
+        # kill all wpa_supplicant relevant to active interface
+
+    def wifi_connect(self, vap: VirtualAPHostapd) -> bool:
+        """Connect to the Access Point. Return True if successful."""
+        config_file_name = "boardfarm_tmp.conf"
+        config_file_path = "/tmp/{}".format(config_file_name)
+
+        # Create network configuration for SSID
+        bssid = "bssid={}".format(vap.bssid)
+        ssid = "ssid=\"{}\"".format(vap.get_ssid())
+        key = "psk=\"{}\"".format(vap.get_psk())
+        network_config = "network={{\n{}\n{}\n{}\n}}".format(bssid, ssid, key)
+        # Clean up previous configuration
+        self.sendline("rm -f \"{}\"".format(config_file_path))
+        self.expect(self.prompt)
+        self.sendline("echo \'{}\' > \"{}\"".format(network_config, config_file_path))
+        self.expect(self.prompt)
+        # Start wpa_supplicant with created configuration
+        # Typical coommand on RPI: wpa_supplicant -B -c/tmp/temp.conf -iwlan0 -Dnl80211,wext
```

Comment:
What about WPS? this assumes you already know the network, meaning after WPS or manual configuration. But in practice, and especially in certification, WPS onboarding always happens so there is no network configuration involved - everything happens as part of WPS.

About the RPI network configuration - does it have a bridge? If so, I suggest to use the `-b` option of the wpa_supplicant (provided it is available in upstream hostapd) so it will manage the STA interface bridge operations.